freeciv-web/src/main/webapp/webclient/index.jsp

```diff
@@ -1,48 +1,47 @@
 <%@ page import="java.util.Properties" %>
 <%@ page import="java.io.IOException" %>
+<%@ page import="static org.apache.commons.lang3.StringUtils.stripToNull" %>
+<%@ page import="static org.apache.commons.lang3.StringUtils.stripToEmpty" %>
+<%@ page import="static java.lang.Boolean.parseBoolean" %>
 <%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
 <%
 String googleSigninClientKey = null;
-String fcwHost = null;
+String trackJsToken = null;
+String minified = "";
 try {
   Properties prop = new Properties();
   prop.load(getServletContext().getResourceAsStream("/WEB-INF/config.properties"));
-  googleSigninClientKey = prop.getProperty("google-signin-client-key");
-  fcwHost = prop.getProperty("fcw_host");
+  googleSigninClientKey = stripToEmpty(prop.getProperty("google-signin-client-key"));
+  trackJsToken = stripToNull(prop.getProperty("trackjs-token"));
+
+  String fcwHost = stripToNull(prop.getProperty("fcw_host"));
+  String debugParam = request.getParameter("debug");
+  boolean hosted = fcwHost != null && request.getServerName().equals(fcwHost);
+  boolean debug = (debugParam != null && (debugParam.isEmpty() || parseBoolean(debugParam)));
+  minified = hosted || !debug ? ".min" : "";
```

Comment:
I'm not sure I understand either the comment or the code. This will only show unminified code when the site is not hosted and the request has the debug param. That is, on hosted site it will always be minified and on dev machine you will have to add the debug parameter. Which may be OK, there's value in testing the minified code (which will be the real one in prod), I just don't think that was the intention given the commit comment.

Reply:
There was a defect in the initial version wrt debug. If you're good relying on just debug though, I rather prefer that, as it makes everything consistent.